jenkins/Jenkinsfile

```diff
@@ -0,0 +1,148 @@
+def secrets = [
+  [path: 'secret/jenkins/github', engineVersion: 2, secretValues: [
+    [envVar: 'PRIVATE_REPO_GITHUB_TOKEN', vaultKey: 'private_token']]],
+]
+
+def configuration = [vaultUrl: "${env.VAULT_URL}",  vaultCredentialId: 'vault-app-role', engineVersion: 2]
+
+def getClusterManagerVersion(String branchName='master'){
+  sh returnStdout:true, script: """
+    eval "\$(curl 'https://raw.githubusercontent.com/cloudify-cosmo/cloudify-cluster-manager/${branchName}/packaging/version_info')"
+    echo "\$CLOUDIFY_CLUSTER_MANAGER_VERSION"
+  """
+}
+
+def getClusterManagerPreRelease(String branchName='master'){
+  sh returnStdout:true, script: """
+    eval "\$(curl 'https://raw.githubusercontent.com/cloudify-cosmo/cloudify-cluster-manager/${branchName}/packaging/version_info')"
+    echo "\$CLOUDIFY_CLUSTER_MANAGER_PACKAGE_RELEASE"
+  """
+}
+
+@Library('pipeline-shared-library') _
+
+pipeline {
+  agent {
+    kubernetes {
+      label 'cluster-manager'
+      defaultContainer 'jnlp'
+      yamlFile 'jenkins/build-pod.yaml'
+    }
+  }
+
+  options {
+    checkoutToSubdirectory('cloudify-cluster-manager')
+    buildDiscarder(logRotator(numToKeepStr:'30'))
+    timeout(time: 60, unit: 'MINUTES')
+    timestamps()
+  }
+
+  environment {
+    BRANCH = "${env.BRANCH_NAME}"
+    PROJECT = "cloudify-cluster-manager"
+    VERSION=getClusterManagerVersion("${env.BRANCH_NAME}").trim()
+    PRERELEASE = getClusterManagerPreRelease("${env.BRANCH_NAME}").trim()
+    S3_BUILD_PATH = "${env.PROJECT}/${env.VERSION}/${env.PRERELEASE}-build/${env.BRANCH}"
```

Comment:
add build number and branch please, to maintain a history, like:    
S3_BUILD_PATH = "${params.VERSION}/${params.PRERELEASE}-build/premium-builder/${params.BRANCH}/${params.BUILD_NUMBER}"

Reply:
Thanks for the comment, I'll add the build number. This package is supposed to be separated from the releases of Cloudify. Therefore, I think the path should be `s3://cloudify-release-eu/cloudify/cloudify-cluster-manager/${params.VERSION}/${params.PRERELEASE}-build/${params.BRANCH}/${params.BUILD_NUMBER}. This is also the reason why I created `getClusterManagerVersion` and `getClusterManagerPreRelease` functions.
Let me know what you think.

Reply:
sounds good!